src/AutoRest.CSharp/Common/Generation/Writers/RequestWriterHelpers.cs

```diff
@@ -258,7 +258,16 @@ private static void WriteHeader(CodeWriter writer, CodeWriterDeclaration request
             using (WriteValueNullCheck(writer, header.Value))
             {
                 writer.Append($"{request}.Headers.{method}({header.Name:L}, ");
-                WriteConstantOrParameter(writer, header.Value, enumAsString: true);
+
+                if (!header.Value.IsConstant && header.Value.Reference.Type.IsFrameworkType && header.Value.Reference.Type.FrameworkType == typeof(Azure.Core.ContentType))
```

Comment:
Not a fan of hardcoding the contentType here. What happens if we don't call ToString on it?

Reply:
> Not a fan of hardcoding the contentType here. What happens if we don't call ToString on it?

The generated code doesn't compile, because there is no conversion from `Azure.Core.ContentType` to `string`.

Reply:
```suggestion
                if (header.Value.Type.Equels(typeof(Azure.Core.ContentType)))
```